### PR TITLE
Update dpkg_lowpkg to handle deb files that have spaces in `dpkg -I` output

### LINKED ISSUE #54739: Unable to use pkg.install with 'sources' parameter in Ubuntu on some deb files
### Description of Issue
Unable to install some deb packages through pkg.install using the 'sources' parameter.

the dpkg_lowpkg module attempts to use 'startswith' to find the information from the output of `dpkg -I ...`

dpkg -I <path/to/deb> returns info with spaces on some packages (McAfee Agent Packages in my case):

```
package.deb
 new Debian package, version 2.0.
 size 1333890 bytes: control archive=1142 bytes.
     260 bytes,     9 lines      control
    1788 bytes,    61 lines   *  postinst             #!/bin/sh
     342 bytes,    18 lines   *  postrm               #!/bin/sh
     401 bytes,    20 lines   *  preinst              #!/bin/sh
     588 bytes,    28 lines   *  prerm                #!/bin/sh
 Package        : <package_name>
 Version        : <version_number>
 Maintainer     : <maintainer>
 Description    : <description>
 Release        : <release_number>
 Architecture   : <arch>
 Section        : <section info>
 Summary        : <summary>
 Pre-Depends    : <depends>
```

The spaces mess up the 'startswith' which uses the 'Architecture:', 'Package:', etc..

### Setup
salt-call pkg.install sources="[{'pkgname':'/path/to/install.deb'}]"

### Steps to Reproduce Issue
salt-call pkg.install sources="[{'pkgname':'/path/to/install.deb'}]"
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 237, in call
    ret['return'] = self.minion.executors[fname](self.opts, data, func, args, kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/modules/aptpkg.py", line 618, in install
    deb_info = __salt__['lowpkg.bin_pkg_info'](pkg_source)
  File "/usr/lib/python2.7/dist-packages/salt/modules/dpkg.py", line 106, in bin_pkg_info
    'Unable to get {0} for {1}'.format(', '.join(missing), path)
CommandExecutionError: Unable to get arch for /var/cache/salt/minion/files/base/install.deb
Error running 'pkg.install': Unable to get arch for /var/cache/salt/minion/files/base/install.deb
```

### Versions Report
```
Salt Version:
           Salt: 2019.2.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15+ (default, Jul  9 2019, 16:51:35)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5

System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-1044-aws
         system: Linux
        version: Ubuntu 18.04 bionic
```

### PR DESCRIPTION
**Pull request by lomeroe**
### What does this PR do?
use re.match to find Name/Architecture/Version instead of startswith, some dpkg -I output has spaces between the description and colon

### What issues does this PR fix or reference?
Fixes #54739 

### Previous Behavior
Deb packages that returned output of `dpkg -I` with spaces between the row name and the ':' would not install

for example
```
# dpkg -I /var/cache/salt/minion/files/base/apps/mcafee/agent/files/MFEdx-ma-5.0.0-568.x86_64.deb
 new Debian package, version 2.0.
 size 1333890 bytes: control archive=1142 bytes.
     260 bytes,     9 lines      control
    1788 bytes,    61 lines   *  postinst             #!/bin/sh
     342 bytes,    18 lines   *  postrm               #!/bin/sh
     401 bytes,    20 lines   *  preinst              #!/bin/sh
     588 bytes,    28 lines   *  prerm                #!/bin/sh
 Package        : MFEdx
 Version        : 5.0.0
 Maintainer     : McAfee, Inc.
 Description    : McAfee Data Exchange Layer
 Release        : 568
 Architecture   : all
 Section        : Network/Agent
 Summary        : McAfee Data Exchange Layer
 Pre-Depends    : MFErt
```

### New Behavior
Packages with this type of output are able to be installed

### Tests written?
Yes

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by xeacott** [comment]
re-run full all

**Comment by xeacott** [comment]
Pesky Windows, its failing like so...
```
[2019-10-03T03:37:40.571Z]         --------  Tests with Errors  ------------------------------------------------------------------------------------------
[2019-10-03T03:37:40.571Z]           -> setUpClass (unit.modules.test_zcbuildout.BuildoutOnlineTestCase)  ................................................
[2019-10-03T03:37:40.571Z]        Traceback (most recent call last):
[2019-10-03T03:37:40.571Z]          File "c:\users\admini~1\appdata\local\temp\kitchen\testing\tests\unit\modules\test_zcbuildout.py", line 330, in setUpClass
[2019-10-03T03:37:40.571Z]            super(BuildoutOnlineTestCase, cls).setUpClass()
[2019-10-03T03:37:40.571Z]          File "c:\users\admini~1\appdata\local\temp\kitchen\testing\tests\unit\modules\test_zcbuildout.py", line 98, in setUpClass
[2019-10-03T03:37:40.571Z]            'setuptools',
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 266, in check_call
[2019-10-03T03:37:40.571Z]            retcode = call(*popenargs, **kwargs)
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 247, in call
[2019-10-03T03:37:40.571Z]            with Popen(*popenargs, **kwargs) as p:
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 676, in __init__
[2019-10-03T03:37:40.571Z]            restore_signals, start_new_session)
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 957, in _execute_child
[2019-10-03T03:37:40.571Z]            startupinfo)
[2019-10-03T03:37:40.571Z]        FileNotFoundError: [WinError 2] The system cannot find the file specified
[2019-10-03T03:37:40.571Z]           .....................................................................................................................
[2019-10-03T03:37:40.571Z]           -> setUpClass (unit.states.test_zcbuildout.BuildoutTestCase)  .......................................................
[2019-10-03T03:37:40.571Z]        Traceback (most recent call last):
[2019-10-03T03:37:40.571Z]          File "C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\kitchen\\testing\tests\unit\modules\test_zcbuildout.py", line 98, in setUpClass
[2019-10-03T03:37:40.571Z]            'setuptools',
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 266, in check_call
[2019-10-03T03:37:40.571Z]            retcode = call(*popenargs, **kwargs)
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 247, in call
[2019-10-03T03:37:40.571Z]            with Popen(*popenargs, **kwargs) as p:
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 676, in __init__
[2019-10-03T03:37:40.571Z]            restore_signals, start_new_session)
[2019-10-03T03:37:40.571Z]          File "C:\Python35\Lib\subprocess.py", line 957, in _execute_child
[2019-10-03T03:37:40.571Z]            startupinfo)
[2019-10-03T03:37:40.571Z]        FileNotFoundError: [WinError 2] The system cannot find the file specified
[2019-10-03T03:37:40.571Z]           .....................................................................................................................
[2019-10-03T03:37:40.571Z]         -----------------------------------------------------------------------------------------------------------------------
[2019-10-03T03:37:40.571Z]        ========================================================================================================================
[2019-10-03T03:37:40.571Z]        FAILED (total=8532, skipped=1126, passed=7404, failures=0, errors=2)
```

I am unsure if this is specific to this build or just the branch 2018.3

**Comment by Akm0d** [comment]
@lomeroe it looks like there are a couple linter errors that need fixed before this can be merged :)

File: tests/unit/modules/test_dpkg_lowpkg.py [hunk 1]
```diff
@@ -12,10 +12,11 @@
 # Import Salt Libs
 import salt.modules.dpkg_lowpkg as dpkg
 
+# Import Salt Testing Libs
 # Import Salt Testing Libs
```
Comment (s0undt3ch) [inline tests/unit/modules/test_dpkg_lowpkg.py:16]: Could you please remove the above 2 comment lines?

File: tests/unit/modules/test_dpkg_lowpkg.py [hunk 2]
```diff
@@ -12,10 +12,11 @@
...
 from tests.support.mixins import LoaderModuleMockMixin
-from tests.support.mock import MagicMock, patch
-from tests.support.unit import TestCase
+from tests.support.mock import NO_MOCK, NO_MOCK_REASON, MagicMock, patch
```
Comment (s0undt3ch) [inline tests/unit/modules/test_dpkg_lowpkg.py:18]: Please leave this import as it was before, ie, remove the `NO_MOCK*` imports

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by xeacott — APPROVED** [review]

**Review by Akm0d — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**